Content.Server/Buckle/Components/StrapComponent.cs

```diff
@@ -36,6 +37,50 @@ public class StrapComponent : SharedStrapComponent, IInteractHand, ISerializatio
         [ViewVariables] [DataField("size")] private int _size = 100;
         private int _occupiedSize;
 
+        /// <summary>
+        /// The buckled entity will be offset by this amount from the center of the strap object.
+        /// If this offset it too big, it will be clamped to <see cref="MaxBuckleDistance"/>
+        /// </summary>
+        [DataField("buckleOffset", required: false)]
+        private Vector2 _buckleOffset = Vector2.Zero;
+
+        private bool _enabled = true;
+
+        /// <summary>
+        /// If disabled, nothing can be buckled on this object, and it will unbuckle anything that's already buckled
+        /// </summary>
+        public bool Enabled
+        {
+            get => _enabled;
+            set
+            {
+                _enabled = value;
+                if (_enabled == value) return;
+                RemoveAll();
+            }
+        }
+
+        /// <summary>
+        /// The distance above which a buckled entity will be automatically unbuckled.
+        /// Don't change it unless you really have to
+        /// </summary>
+        [DataField("maxBuckleDistance", required: false)]
+        public float MaxBuckleDistance = 1f;
+
+        /// <summary>
+        /// You can specify the offset the entity will have after unbuckling.
+        /// </summary>
+        [DataField("unbuckleOffset", required: false)]
+        public Vector2 UnbuckleOffset = Vector2.Zero;
```

Comment:
Seems like it'd be better to just have unbuckle leave them at their current worldposition instead?

Reply:
In some cases this is useful, for example for beds, if you stay in the same position it pushes the bed around and we do not want that

Reply:
But if we collide with beds then that means if the bed is ever next to a wall this won't work though? We'll just get stuck like we used to with disposals.